os/drivers/ai-soc/ndp120/src/ndp120_debug_utils.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2024 Syntiant Corp.  All rights reserved.
+ * Contact at http://www.syntiant.com
+ *
+ * This software is available to you under a choice of one of two licenses.
+ * You may choose to be licensed under the terms of the GNU General Public
+ * License (GPL) Version 2, available from the file LICENSE in the main
+ * directory of this source tree, or the OpenIB.org BSD license below.  Any
+ * code involving Linux software will require selection of the GNU General
+ * Public License (GPL) Version 2.
+ *
+ * OPENIB.ORG BSD LICENSE
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+*/
+
+#include <tinyara/kmalloc.h>
+#include <tinyara/wqueue.h>
+#include <tinyara/pm/pm.h>
+
+#include <assert.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <signal.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <errno.h>
+
+#include "../../../audio/ndp120_voice.h"
+#include <syntiant_ilib/syntiant_ndp_error.h>
+
+static int pdm_clk_en(struct syntiant_ndp_device_s *ndp, uint32_t clk)
+{
+	struct syntiant_ndp120_config_pdm_s pdm_config;
+	int s;
+
+	memset(&pdm_config, 0, sizeof(pdm_config));
+	pdm_config.interface = 0;
+	pdm_config.clk = clk;
+
+	pdm_config.set = SYNTIANT_NDP120_CONFIG_SET_PDM_CLK;
+	s = syntiant_ndp120_config_pdm(ndp, &pdm_config);
+	if (s) {
+		auddbg("PDM clock set (%d) failed\n", clk);
```

Comment:
Don't you need to print the reason of failure using s?
same in other failure cases